src/stakeinput.h

```diff
@@ -35,14 +35,14 @@ class CStakeInput
 class CPivStake : public CStakeInput
 {
 private:
-    CTransaction txFrom{CTransaction()};
-    unsigned int nPosition{0};
+    Optional<CTxOut> opOutputFrom{nullopt};
+    Optional<COutPoint> opOutpointFrom{nullopt};
```

Comment:
(f55cfbc3cca51deee5b4e645b6c0bf19c7f20a2c)
Many places are accessing these fields without first checking if they are initialized (and we can never be sure, until `SetPrevout`). Maybe would be better to use empty `CTxOut` and `COutPoint`, rather than optionals.